parsl/monitoring/monitoring.py

```diff
@@ -122,22 +123,23 @@ def send(self, message_type, task_id, message):
         return x
 
 
+@typeguard.typechecked
 class MonitoringHub(RepresentationMixin):
     def __init__(self,
-                 hub_address,
-                 hub_port=None,
-                 hub_port_range=(55050, 56000),
-
-                 client_address="127.0.0.1",
-                 client_port_range=(55000, 56000),
-
-                 workflow_name=None,
-                 workflow_version=None,
-                 logging_endpoint='sqlite:///monitoring.db',
-                 logdir=None,
-                 monitoring_debug=False,
-                 resource_monitoring_enabled=True,
-                 resource_monitoring_interval=30):  # in seconds
+                 hub_address: str,
+                 hub_port: Optional[int] = None,
+                 hub_port_range: Tuple[int, int] = (55050, 56000),
+
+                 client_address: str = "127.0.0.1",
+                 client_port_range: Tuple[int, int] = (55000, 56000),
+
+                 workflow_name: Optional[str] = None,
+                 workflow_version: Optional[str] = None,
+                 logging_endpoint: str = 'sqlite:///monitoring.db',
+                 logdir: Optional[str] = None,
+                 monitoring_debug: bool = False,
+                 resource_monitoring_enabled: bool = True,
+                 resource_monitoring_interval: int = 30):  # in seconds
```

Comment:
I think this `resource_monitoring_interval` should be float (the docstrings were wrong before this PR)

Reply:
I've change it to float both here and in the docstring.